packages/flutter/lib/src/rendering/box.dart

```diff
@@ -210,21 +210,23 @@ class BoxConstraints extends Constraints {
   /// as close as possible to the original constraints.
   BoxConstraints enforce(BoxConstraints constraints) {
     return BoxConstraints(
-      minWidth: minWidth.clamp(constraints.minWidth, constraints.maxWidth),
-      maxWidth: maxWidth.clamp(constraints.minWidth, constraints.maxWidth),
-      minHeight: minHeight.clamp(constraints.minHeight, constraints.maxHeight),
-      maxHeight: maxHeight.clamp(constraints.minHeight, constraints.maxHeight),
+      minWidth: minWidth.clamp(constraints.minWidth, constraints.maxWidth) as double,
```

Comment:
I wish clamp on a double would just always return a double. 

Maybe we can add an extension method `double clamp(double lowerLimit, double upperLimit)` to foundation that implements that protocol instead of casting the clamp everywhere? It seems cluttery...

Reply:
>Maybe we can add an extension method

That's what I planned to do in an upcoming PR (to avoid too much changes).

However I don't think `clamp` can be used as name. I was thinking to `clampDouble` and `clampInt`. (Note that the SDK could also make some changes  https://github.com/dart-lang/sdk/issues/30403#issuecomment-323717772 )

Reply:
Another name for the function could be `within`.

Reply:
I would prefer it if we did the clamp extension method before the implicit-cast:false change then, since the clamp stuff adds a lot of noise to them..

Reply:
>I would prefer it if we did the clamp extension method before the implicit-cast:false change then, since the clamp stuff adds a lot of noise to them

Regarding all the changes I already made it would be a nightmare to handle merge conflits if I change `clamp` before implicit-casts now :-( Don't worry the extension clamp changes should be really easy/quick to do afterwards.